Add Advanced examples to Documentation

JMuff22 opened the pull request:
Hi, I would like to add 2 examples to the cirq-on-iqm docs: 
1. Batch submission with `IQMSampler.run_iqm_batch`
2. Parameterized submission using ` cirq.Linspace` + `IQMSampler.run_sweep`

I can also add these examples to the `examples` directory. 

I noticed that the documentation is somewhat formatted, however tox/black does not auto format it. Please also comment on the line lengths and how previous authors have kept it consistent. 

I would also like to add a table similar to https://iqm-finland.github.io/qiskit-on-iqm/user_guide.html#running-a-quantum-circuit-on-an-iqm-quantum-computer showing the different backend options. Why were these previously not added? They could be added for example under `Running on a real quantum computer`. 

Thanks!

q-mat-beu commented on docs/user_guide.rst:
```diff
@@ -290,8 +290,52 @@ When executing a circuit that uses something other than the device qubits, you n
+
+In this section we demonstrate some of the more advanced examples of using Cirq on IQM and how you can use various tools available in Cirq.
+
+Multiple circuits can be submitted to the IQM quantum computer at once using the :meth:`~.IQMSampler.run_iqm_batch` method of :class:`.IQMSampler`. 
+
+.. code-block:: python
+   
+   circuit_list = []
+
+   circuit_list.append(routed_circuit_1)
+   circuit_list.append(routed_circuit_2)
+
+   results = sampler.run_iqm_batch(circuit_list, repetitions=10)
+
+   for result in results:
+        print(result.histogram(key="m"))
```
This would fit better under the section `Running on a real quantum computer` with a note on performance improvements when running circuits in a batch.

JMuff22 replied:
Resolved in [76f994b](https://github.com/iqm-finland/cirq-on-iqm/pull/117/commits/76f994bc08e2c3f7e6a3351ac8cbb31d13ebc402)

q-mat-beu commented on docs/user_guide.rst:
```diff
@@ -290,8 +290,52 @@ When executing a circuit that uses something other than the device qubits, you n
+        print(result.histogram(key="m"))
+
+
+Similiarly, parameterized circuits can be executed using :meth:`~.IQMSampler.run_sweep` and :meth:`~.cirq.resolve_parameters`.
+
+.. code-block:: python
+   
+   import sympy
+   q1, q2 = cirq.NamedQubit('Alice'), cirq.NamedQubit('Bob')
+   theta = sympy.Symbol("theta")
+   circuit = cirq.Circuit([
+    cirq.H(q1),
+    cirq.CNOT(q1, q2),
+    cirq.Z(q1) ** theta,
+    cirq.Z(q2) ** theta,
+    cirq.CNOT(q1, q2),
+    cirq.H(q1),
+    cirq.measure(q1, q2, key='m'),
+   ])
+
+   decomposed_circuit = adonis.decompose_circuit(circuit)
+   routed_circuit, _, _ = adonis.route_circuit(decomposed_circuit)
+   simplified_circuit = simplify_circuit(routed_circuit)
+
+   params = cirq.Linspace(key="theta", start=0, stop=0.5, length=3)
+   results = sampler.run_sweep(simplified_circuit, repetitions=10, params=params)
+   for result in results:
+       print(f'{result.params}: {result.histogram(key="m")}')
+
```
We have explicitly omitted documentation that is related to core `cirq` features like this one to avoid duplicating information and having to maintain this in sync with `cirq` docs.

JMuff22 replied:
Removed in [76f994b](https://github.com/iqm-finland/cirq-on-iqm/pull/117/commits/76f994bc08e2c3f7e6a3351ac8cbb31d13ebc402)

q-mat-beu reviewed (changes requested):
Hi @JMuff22 and thanks for the initiative!

I think 1) is a valuable addition but I would omit 2) for the reason explained in my comment.

As for the table I think this a valuable addition, feel free to add it as well.

Regarding the formatting we don't have tooling yet unfortunately, please try and stick to the existing style.

Thanks!

JMuff22 commented:
Hi @q-mat-beu  thanks for the feedback, please see the changes!

> As for the table I think this a valuable addition, feel free to add it as well.

Added in [5cb9c34](https://github.com/iqm-finland/cirq-on-iqm/pull/117/commits/5cb9c34a8440880b21856ec8befcb1ff2f94ff77) is the `IQMSampler` options table. I also edited the text, for example:

1. `result = sampler.run(circuit, repetitions=10, circuit_duration_check=False)` no longer works and should be removed. `circuit_duration_check` is an option of `IQMSampler` in my testing. 
2. Removed redundant text that is already included in the table.

q-mat-beu commented on docs/user_guide.rst:
```diff
@@ -262,21 +262,45 @@ instance and use its :meth:`~.IQMSampler.run` method to send a circuit for execu
+   * - `heralding_mode`
+     - :py:class:`~iqm_client.iqm_client.HeraldingMode`
+     - "zeros"
+     - Heralding mode to use during execution. The default value is "none".
```
```suggestion
     - Heralding mode to use during execution. The default value is "none", "zeros" enables heralding.
```

q-mat-beu reviewed (changes requested):
Hi @JMuff22 , looks good, thanks! If you can add the `CHANGELOG.rst` entry we are ready to merge!

q-mat-beu commented on CHANGELOG.rst:
```diff
@@ -2,6 +2,11 @@
+Version 11.13
+=============
+
+* Add table of backend options to the user guide. Add example of submitting batch of circuits to the user guide. `#117 <https://github.com/iqm-finland/cirq-on-iqm/pull/117>`_
```
```suggestion
* Add table of backend options and an example of submitting a batch of circuits to the user guide. `#117 <https://github.com/iqm-finland/cirq-on-iqm/pull/117>`_
```
